src/js/modules/supervisors/Repository.ts

```diff
@@ -124,6 +124,7 @@ class Repository {
             for (const [key, value] of resultRecordBatch) {
               value.records = value.records.map((record) => {
                 record.length = calculateRecordLength(record);
+                record.valueLen = record.value.length;
```

Comment:
Why is there a `valueLen` field if you can just get `record.value.length`?

Reply:
because `valueLen` is the buffer size for record value.